utils/cloud_metadata.py

```diff
@@ -14,6 +14,36 @@
 
 log = logging.getLogger(__name__)
 
+class Azure(object):
+    URL = "http://169.254.169.254/metadata/instance?api-version=2017-04-02"
+    TIMEOUT = 0.3 # second
+
+    @staticmethod
+    def _get_metadata(agentConfig):
+        if not agentConfig.get('collect_instance_metadata', True):
+            log.info("Instance metadata collection is disabled. Not collecting it.")
```

Comment:
nit: could you make the message here more specific, `Not collecting Azure metadata` for example :)